price_feeder/tasks.py

```diff
@@ -366,40 +370,38 @@ def post_price(self, price_no_precision, info_contracts, post, re_post, task=Non
-                # we need to check if the previous tx, the one that we want to replace in first place,
-                # is not already mined. If we replaced a tx, but the previous one was mined at the same time
-                # the new one stay pending forever and will be tried to be replaced always.
-                # for example,
-                # post condition -> create tx0 with nonce 100 = pending
-                # re post condition -> create tx1 with nonce 100 = pending  ----> at the same time, tx0 nonce 100 = confirmed
-                # re post condition -> tx2 with nonce 100 is not created because tx0 is confirmed, so return
-                # without this check, tx2 will be created because is reading tx1 pending condition but will always revert 
-                # because tx0 is already confirmed with the same nonce 100
-                try:
-                    tx_rcp = chain.get_transaction(global_manager['tx0'])
-                # if the tx0 is not found return just in case
-                except exceptions.TransactionNotFound:
-                    return
-                # check if the tx0 is mined, if it is return
-                if tx_rcp.nonce == global_manager['last_used_nonce'] and tx_rcp.confirmations >= 1: 
-                    return       
-                last_used_nonce = global_manager['last_used_nonce']
-                calculated_gas_price = decimal.Decimal(global_manager['last_gas_price']) + decimal.Decimal(Web3.fromWei(self.options['networks'][self.config_network]['re_post_gas_price_increment'], 'ether'))
-
-                                
-                log.info(" Replacing tx price "
-                    "New gas price: [{0:.18f}] "
-                    "Nonce: [{1}] ".format(
-                calculated_gas_price,
-                last_used_nonce,
-                ))
+            # node just start and doesn't have previous information
+            if not ('tx0' in global_manager and 'last_used_nonce' in global_manager and 'last_gas_price' in global_manager): return
+            # we need to check if the previous tx, the one that we want to replace in first place,
+            # is not already mined. If we replaced a tx, but the previous one was mined at the same time
+            # the new one stay pending forever and will be tried to be replaced always.
+            # for example,
+            # post condition -> create tx0 with nonce 100 = pending
+            # re post condition -> create tx1 with nonce 100 = pending  ----> at the same time, tx0 nonce 100 = confirmed
+            # re post condition -> tx2 with nonce 100 is not created because tx0 is confirmed, so return
+            # without this check, tx2 will be created because is reading tx1 pending condition but will always revert
+            # because tx0 is already confirmed with the same nonce 100
+            try:
+                tx_rcp = chain.get_transaction(global_manager['tx0'])
```

Comment:
What happens if it's not tx0 nor tx2 the one confirmed, but tx1, let's suppose this scenario:
tx0: discarded as it was replace by tx1
tx1: confirmed (but we didn't saw it)
tx2: broadcasted, as when we checked, tx1 was not still confirmed
Now we are evaluating re-posting tx3, would we recognize we shouldn't? And more important, can we avoid the deadlock condition?

Reply:
but once tx1 is broadcasted => tx0 = tx1 and will be checked before broadcast tx2